src/components/AddEvent.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class AddEvent extends Component {
+  render() {
+    const failed = !this.props.addEventResponse.addSuccess && this.props.addEventResponse.addSuccess !== undefined;
+    return (
+      <div>
+        <form onSubmit={event => this.props.onFormSubmit(event, this.props.addEventFields)}>
+          <div>
+            <label htmlFor="eventName">Name:</label>
+            <input
+              type="text"
+              id="eventName"
+              name="event_name"
+              onChange={this.props.onEventNameInput}
+            />
+          </div>
+          <div>
+            <label htmlFor="artist">Artist</label>
+            <input
+              type="text"
+              id="artist"
+              name="artist_name"
+              onChange={this.props.onArtistInput}
+            />
+          </div>
+          <div>
+            <label htmlFor="venue">Venue</label>
+            <input
+              type="text"
+              id="venue"
+              name="venue_name"
+              onChange={this.props.onVenueInput}
+            />
+          </div>
+          <div>
+            <label htmlFor="address">Address</label>
+            <input
+              type="text"
+              id="address"
+              name="address"
+              onChange={this.props.onAddressInput}
+            />
+          </div>
+          <div>
+            <label htmlFor="price">Price</label>
+            <input
+              type="text"
+              id="price"
+              name="price"
+              onChange={this.props.onPriceInput}
+            />
+          </div>
+          <div>
+            <label htmlFor="start">Opening Time:</label>
+            <input
+              type="text"
+              id="start"
+              name="opening_time"
+              onChange={this.props.onStartTimeInput}
+            />
+          </div>
+          <div>
+            <label htmlFor="end">Closing Time</label>
+            <input
+              type="text"
+              id="end"
+              name="closing_time"
+              onChange={this.props.onEndTimeInput}
+            />
+          </div>
+          <div className="button">
+            <button type="submit" >Create</button>
+          </div>
+          { failed ? <p>WOMP!</p> : <div /> }
+        </form>
+      </div>
+    );
+  }
+}
+
+AddEvent.propTypes = {
+  onFormSubmit: PropTypes.func.isRequired,
+  onEventNameInput: PropTypes.func.isRequired,
+  onArtistInput: PropTypes.func.isRequired,
+  onVenueInput: PropTypes.func.isRequired,
+  onAddressInput: PropTypes.func.isRequired,
+  onPriceInput: PropTypes.func.isRequired,
+  onStartTimeInput: PropTypes.func.isRequired,
+  onEndTimeInput: PropTypes.func.isRequired,
+  addEventResponse: PropTypes.shape(),
+  addEventFields: PropTypes.shape().isRequired,
+};
+
+AddEvent.defaultProps = {
+  addEventResponse: {},
```

Comment:
@GameKyuubi I'm not familiar with defaultProps. Could you explain why you need this one?

Reply:
@kumiko-haraguchi If a prop is undefined by default, `.isRequired` doesn't work because it demands that the prop is never undefined.  If we omit `.isRequired` then it allows the prop to be undefined, but then requires the prop to be set manually in `defaultProps`.